torch/csrc/jit/runtime/register_prim_ops.cpp

```diff
@@ -2192,6 +2193,44 @@ int hashValue(Stack& stack) {
   return 0;
 }
 
+int64_t getIValueId(const IValue& val) {
+  // id in practice is only used to compare pointer types, and None.
+  // Python gives no guarantees otherwise.
+  if (val.isPtrType()) {
+    return reinterpret_cast<int64_t>(val.internalToPointer());
+  }
+  if (val.isNone()) {
```

Comment:
I am confused about the goals of the below code. What invariant is it trying to maintain? Possible invariants include:`id(x) == id(y) implies x == y` but this is not maintained.  E.g. an integer with its top bit already set is equal to the one with out. One simpler option would just be to return -1 for all non-objects.